### PR TITLE
DTSPO-5341 Launch Darkly documentation

### PR DESCRIPTION
**Pull request by mokainos**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: source/ways-of-working/new-component/LaunchDarkly.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+title: LaunchDarkly
```
Comment (timja) [inline source/ways-of-working/new-component/LaunchDarkly.html.md.erb:2]:
```suggestion
title: Feature flags
```

File: source/ways-of-working/new-component/LaunchDarkly.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+title: Feature flags
```
Comment (timja) [inline source/ways-of-working/new-component/LaunchDarkly.html.md.erb:2]: name change applies to file name and other link

File: source/ways-of-working/new-component/LaunchDarkly.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+weight: 99
+---
+
+# <%= current_page.data.title %>
+
+Feature flags are a software development and delivery technique that allows software teams to enable and disable parts of a codebase at the flip of a switch.
+Feature flags enable software teams to adopt DevOps practices, in so much as they help increase deployment speeds, system stability, and cross-team collaboration.
+Feature flags also allow you to decouple code deployments from feature releases. You can make code changes in production while hiding those changes from users.
+
+### Accessing LaunchDarkly.
```
Comment (timja) [inline source/ways-of-working/new-component/LaunchDarkly.html.md.erb:14]: Needs to include how to setup a new project:
https://github.com/hmcts/launchdarkly-terraform

and should have a section on Environments, how we only have the testing and production environments. generally flags should be globally set the same as in production and user targeting should be used apart from that.

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+weight: 99
+---
+
+# <%= current_page.data.title %>
+
+Feature flags are a software development and delivery technique that allows software teams to enable and disable parts of a codebase at the flip of a switch.
+Feature flags enable software teams to adopt DevOps practices, in so much as they help increase deployment speeds, system stability, and cross-team collaboration.
+Feature flags also allow you to decouple code deployments from feature releases. You can make code changes in production while hiding those changes from users.
+
+### Accessing LaunchDarkly.
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:14]: headings aren't sentences no need to add a `.`
```suggestion
### Accessing LaunchDarkly
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    - Developers automatically get access by being part of 'DTS CFT Developers' or 'DTS SDS Developers', non developers can get access with the 'DTS LaunchDarkly' group.
+      see [onboarding](../../onboarding/person/index.html) process to request access.
+    - [URL](https://app.launchdarkly.com/)
+
+### Documentation.
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:20]:
```suggestion
### Documentation
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    - Documentation can be accessed from directory from [LaunchDarkly](https://docs.launchdarkly.com/home) website.
+    - [Blog](https://launchdarkly.com/blog/) contain rich source of information.
+
+### setup.
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:25]:
```suggestion
### Creating a new project
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+    [LaunchDarkly](https://github.com/hmcts/launchdarkly-terraform/blob/74156b2b71ba4709e289d534fc346733300014a1/terraform.tfvars#L6) terraform module is used to create the project.
+
+### Enviornments.
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:29]:
```suggestion
### Environments
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+    We currently target 'Test' and 'Production' enviornments only. Flags are set generally the same as in 'production' enviornment and use flag variation targeting users.
+
+### Housekeeping.
+
+    As part of housekeeping, remove corresponding project entries from [tfvars](https://github.com/hmcts/launchdarkly-terraform/blob/74156b2b71ba4709e289d534fc346733300014a1/terraform.tfvars#L6) file
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:35]: not needed
```suggestion
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    - [URL](https://app.launchdarkly.com/)
+
+### Documentation.
+
+    - Documentation can be accessed from directory from [LaunchDarkly](https://docs.launchdarkly.com/home) website.
+    - [Blog](https://launchdarkly.com/blog/) contain rich source of information.
+
+### setup.
+
+    [LaunchDarkly](https://github.com/hmcts/launchdarkly-terraform/blob/74156b2b71ba4709e289d534fc346733300014a1/terraform.tfvars#L6) terraform module is used to create the project.
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:27]:
```suggestion
    See [hmcts/launchdarkly-terraform](https://github.com/hmcts/launchdarkly-terraform) for instructions on how to setup your new project.
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+### Enviornments.
+
+    We currently target 'Test' and 'Production' enviornments only. Flags are set generally the same as in 'production' enviornment and use flag variation targeting users.
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:31]: maybe this?
```suggestion
    We currently provide 'Test' and 'Production' environments only within Launch Darkly. Flags are generally set the same value in both 'Production' and the 'Test' environment, flag targeting is used to control rollout of flags to users across environments.
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+### Accessing LaunchDarkly.
+
+    - Developers automatically get access by being part of 'DTS CFT Developers' or 'DTS SDS Developers', non developers can get access with the 'DTS LaunchDarkly' group.
+      see [onboarding](../../onboarding/person/index.html) process to request access.
+    - [URL](https://app.launchdarkly.com/)
+
+### Documentation.
+
+    - Documentation can be accessed from directory from [LaunchDarkly](https://docs.launchdarkly.com/home) website.
```
Comment (njrich28) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:22]:
```suggestion
    - Documentation can be accessed directly from the [LaunchDarkly](https://docs.launchdarkly.com/home) website.
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+last_reviewed_on: 2021-10-29
+review_in: 6 months
+weight: 99
+---
+
+# <%= current_page.data.title %>
+
+Feature flags are a software development and delivery technique that allows software teams to enable and disable parts of a codebase at the flip of a switch.
+Feature flags enable software teams to adopt DevOps practices, in so much as they help increase deployment speeds, system stability, and cross-team collaboration.
+Feature flags also allow you to decouple code deployments from feature releases. You can make code changes in production while hiding those changes from users.
```
Comment (njrich28) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:12]: Maybe include that it's an enabler of trunk-based development? https://trunkbaseddevelopment.com/

**Review by njrich28 — COMMENTED** [review]
This is good 👍  might be nice to include a diagram of the flow somewhere - i've got a starter diagram (that could do with some improvement) if it helps
Screenshot 2021-11-01 at 10 17 42
e

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+title: Feature flags
+last_reviewed_on: 2021-10-29
+review_in: 6 months
+weight: 99
+---
+
+# <%= current_page.data.title %>
+
+Feature flags are a software development and delivery technique that allows software teams to enable and disable parts of a codebase at the flip of a switch.
+Feature flags enable software teams to adopt DevOps practices, in so much as they help increase deployment speeds, system stability, and cross-team collaboration.
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:11]: I don't think DevOps practices is what I would say here,

```suggestion
Feature flags enable a team to deliver new features to use rapidly but safely.
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+Feature flags enable a team to deliver new features to use rapidly but safely.
+Feature flags also allow you to decouple code deployments from feature releases. You can make code changes in production while hiding those changes from users.
+
+### Trunk Based Development
+Trunk Based Development is a branching model in which developers create short-lived feature branches and merge back into the “trunk” branch, often called as the master/main branch.
+It reduces the complexity of merging events and keeps code current by having fewer development lines and by doing small and frequent merges.
+Trunk based Development and Feature Flags together can be used for continuous delivery, delivering the features faster to market.
+Trunk Based Development uses Feature Flags as a mechanism to manage new feature releases. Feature flags allow developers to deploy incomplete features to production without releasing them to end-users.
+
+![TBD with Feature Flag](/images/TBD_FeatureFlag.png)
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:20]:
```suggestion
![Trunk based development with feature flags](/images/TBD_FeatureFlag.png)
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+### Accessing LaunchDarkly
+
+    - Developers automatically get access by being part of 'DTS CFT Developers' or 'DTS SDS Developers', non developers can get access with the 'DTS LaunchDarkly' group.
+      see [onboarding](../../onboarding/person/index.html) process to request access.
+    - [URL](https://app.launchdarkly.com/)
+
+### Documentation
+
+    - Documentation can be accessed directly from the [LaunchDarkly](https://docs.launchdarkly.com/home) website.
+    - [Blog](https://launchdarkly.com/blog/) contain rich source of information.
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:39]:
```suggestion
    - [Blog](https://launchdarkly.com/blog/) contains good information.
```

File: source/ways-of-working/new-component/feature-flags.html.md.erb [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+### Creating a new project
+
+    See [hmcts/launchdarkly-terraform](https://github.com/hmcts/launchdarkly-terraform) for instructions on how to setup your new project.
+
+### Environments
+
+    We currently provide 'Test' and 'Production' environments only within Launch Darkly. Flags are generally set the same value in both 'Production' and the 'Test' environment, flag targeting is used to control rollout of flags to users across environments.
```
Comment (timja) [inline source/ways-of-working/new-component/feature-flags.html.md.erb:47]:
```suggestion
    We currently provide 'Test' and 'Production' environments only within Launch Darkly. Flags are generally set to the same value in both 'Production' and the 'Test' environment, flag targeting is used to control rollout of flags to users across environments.
```

**Review by timja — APPROVED** [review]

**Comment by zc-hmcts** [comment]
Screenshot 2021-11-09 at 15 19 40
links look wrong

**Pull request merged**